pkg/workloads/console/integration/integration_test.go

```diff
@@ -66,37 +88,211 @@ var _ = Describe("Console", func() {
+			Expect(mgr.GetClient().Create(context.TODO(), consoleTemplate)).NotTo(
+				HaveOccurred(), "failed to create Console Template",
+			)
+
+			csl := &workloadsv1alpha1.Console{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "my-first-console",
+					Namespace: namespace,
+				},
+				Spec: workloadsv1alpha1.ConsoleSpec{
+					ConsoleTemplateRef: corev1.LocalObjectReference{Name: "my-first-console-template"},
+				},
+			}
+
+			By("Creating console")
+			Expect(mgr.GetClient().Create(context.TODO(), csl)).NotTo(
+				HaveOccurred(), "failed to create Console",
+			)
+
+			By("Expect reconcile succeeded")
+			Eventually(calls, timeout).Should(
+				Receive(
+					integration.ReconcileResourceSuccess(namespace, "my-first-console"),
+				),
+			)
+
+			By("Expect job was created")
+			job := &batchv1.Job{}
+			identifier, _ := client.ObjectKeyFromObject(csl)
+			err := mgr.GetClient().Get(context.TODO(), identifier, job)
+
+			Expect(err).NotTo(HaveOccurred(), "failed to find associated Job for Console")
+			// Expect(job.Spec.Template).To(Equal(consoleTemplate.Spec.Template), "the job's pod spec should match the console template")
+			Expect(job.Spec.Template.Spec.Containers[0].Image).To(Equal("alpine:latest"), "the job's pod runs the same container as specified in the console template")
+			// TODO: Test for correct logs
+		})
+
+		It("Only creates one job when reconciling twice", func() {
+			consoleTemplate := &workloadsv1alpha1.ConsoleTemplate{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "my-first-console-template",
+					Namespace: namespace,
+				},
+				Spec: workloadsv1alpha1.ConsoleTemplateSpec{
+					AdditionalAttachSubjects: []rbacv1.Subject{},
+					Template: corev1.PodTemplateSpec{
+						Spec: corev1.PodSpec{
+							Containers: []corev1.Container{
+								corev1.Container{
+									Image:   "alpine:latest",
+									Name:    "console-container-0",
+									Command: []string{"sleep", "100"},
+								},
+							},
+							RestartPolicy: "Never",
+						},
+					},
+				},
+			}
+
+			By("Creating console template")
+			Expect(mgr.GetClient().Create(context.TODO(), consoleTemplate)).NotTo(
+				HaveOccurred(), "failed to create Console Template",
+			)
+			csl := &workloadsv1alpha1.Console{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "my-first-pod",
+					Namespace: namespace,
+				},
+				Spec: workloadsv1alpha1.ConsoleSpec{
+					ConsoleTemplateRef: corev1.LocalObjectReference{Name: "my-first-console-template"},
+				},
+			}
+
+			By("Creating console")
+			Expect(mgr.GetClient().Create(context.TODO(), csl)).NotTo(
+				HaveOccurred(), "failed to create Console",
+			)
+
+			By("Expect reconcile succeeded")
+			Eventually(calls, timeout).Should(
+				Receive(
+					integration.ReconcileResourceSuccess(namespace, "my-first-pod"),
+				),
+			)
+
+			By("Reconciling again")
+			identifier, _ := client.ObjectKeyFromObject(csl)
+			go func() { consoleController.Reconcile(reconcile.Request{identifier}) }()
```

Comment:
What was your thought here on calling Reconcile directly instead of updating the resource and expecting the result via the channel?

Reply:
I had thought that it was nicer to not to a pointless update to the resource just to test reconciliation, but on reflection I think it actually makes more sense to do that, as it's in line with the kind of user action that would cause a reconciliation.